dds/DCPS/DataWriterImpl_T.h

```diff
@@ -427,10 +431,13 @@ namespace OpenDDS {
         if (key_marshaled_size_) {
           effective_size = key_marshaled_size_;
         } else {
-          if (cdr) {
-            effective_size = 4; // CDR encapsulation
+          if (cdr && !Serializer::use_rti_serialization()) {
+            effective_size = cdr_header_size; // CDR encapsulation
           }
           TraitsType::gen_find_size(ko_instance_data, effective_size, padding);
+          if (Serializer::use_rti_serialization()) {
+            effective_size += (8 - cdr_header_size);  // RTI padding
```

Comment:
This can be simply "cdr_header_size".  The difference is that the effective_size is incremented *after* gen_find_size so that gen_find_size doesn't consider the CDR header when determining alignment.